[DOCU-2776] Consumer Groups for Konnect (#4911)
Kong/docs.konghq.com: app/konnect/runtime-manager/consumer-groups.md

---
title: Create Consumer Groups in Konnect

> content_type: tutorial

Reviewer: I think this could be a how-to and it may be easier to strip some of the information. Otherwise, if you want to keep this to a tutorial you should write a concluding paragraph explaining what was done and how it can be changed.

badge: enterprise
---
With consumer groups, you can define any number of rate limiting tiers and
apply them to subsets of consumers, instead of managing each consumer
individually.

For example, you could define three consumer groups:
* A "gold tier" with 1000 requests per minute
* A "silver tier" with 10 requests per second
* A "bronze tier" with 6 requests per 30 seconds

The `consumer_groups` endpoint works together with the [Rate Limiting Advanced plugin](/hub/kong-inc/rate-limiting-advanced).

Consumers that are not in a consumer group default to the Rate Limiting advanced
plugin’s configuration, so you can define tier groups for some users and
have a default behavior for consumers without groups.

To use consumer groups for rate limiting, you must:
* Create one or more consumer groups

> * Create consumers

Reviewer: Can we merge these steps? Like "Create consumers and add them to an already exisisting consumer group", or "Create a consumer group and create or add consumers to it"

* Assign consumers to groups
* Configure the Rate Limiting Advanced plugin globally with the `enforce_consumer_groups`
and `consumer_groups` parameters, setting up the list of consumer groups that
the plugin accepts
* Configure a rate limiting policy for each consumer group, overriding the 
plugin's global configuration

## Set up a consumer group with consumers

In this section, you will create the Bronze tier consumer group with a consumer assigned to the group.

1. In {{site.base_short_name}}, click **Runtime Manager** in the sidebar.

> 1. Click the **default** runtime group.

Reviewer: ```suggestion
1. Select the **default** runtime group.
```

> 1. From the menu, click **Consumers**, then click **Add consumer**.

Reviewer: ```suggestion
1. Open the **Consumers** tab, click **Add consumer**, and enter the consumer information into the form.
```

> 1. Enter a **Username** and **Custom ID**. For this example, you can use "Amal" for each field.

Reviewer: ```suggestion

```

> 1. Click **Save**.

Reviewer: ```suggestion
```

> 1. Click the **Consumer Groups** tab.

Reviewer: I think you should split these two instruction sets. The first one created a consumer, and should belong in an H2 about creating a consumer, and the next one adds it to a group.

> 1. Click **Add Consumer Group**.

Reviewer: Maybe merge step 1 + 2

1. Enter "Bronze" for the consumer group name and select the "Amal" consumer you just created from the drop-down. 
1. Click **Create**.

## Configure the Rate Limiting Advanced plugin for all consumers

In this section, you will configure the Rate Limiting Advanced plugin to set the rate limit to 5 requests every 30 seconds for all consumers.

> 1. In {{site.base_short_name}}, click **Runtime Manager** in the sidebar.

Reviewer: ```suggestion
1. In {{site.konnect_short_name}}, click **Runtime Manager** in the sidebar.
```

1. Click the **default** runtime group.
1. From the menu, open **Plugins**, then click **Add plugin**.
1. Find the **Rate Limiting** plugin, then click **Select**.

> 1. Apply the plugin as **Global**, which means the rate limiting applies to all requests, including every service and route in the workspace.

Reviewer: ```suggestion
1. Apply the plugin as **Global**
    This means the rate-limiting applies to all requests, including every service and route in the workspace.
```

    If you switched it to **Scoped**, the rate limiting would apply the plugin to only one service, route, or consumer.

    By default, the plugin is automatically enabled when the form is submitted.
    You can also toggle the **This plugin is Enabled** button to configure the plugin without enabling it.
    For this example, keep the plugin enabled.
1. Complete only the following fields with the following parameters:
    1. Config.limit: `5`
    1. Config.window_size: `30`
    1. Config.window_type: `Sliding`
    1. Config.retry_after_jitter_max: `0`
    1. Config.enforce_consumer_groups: `true` 
    1. Config.consumer_groups: `Bronze`

    Besides the above fields, there may be others populated with default values. For this example, leave the rest of the fields as they are.
1. Click **Save**.

## Configure rate limiting for consumer groups

In this section, you will configure the Rate Limiting Advanced plugin to set the rate limit to 6 requests every 30 seconds only for consumers in the Bronze tier.

> 1. In {{site.base_short_name}}, click **Runtime Manager** in the sidebar.

Reviewer: ```suggestion
1. In {{site.konnect_short_name}}, click **Runtime Manager** in the sidebar.
```

> 1. Click the **default** runtime group.

Reviewer: ```suggestion
1. Select the **default** runtime group.
```

1. From the menu, open **Consumers**, then click the **Consumer Groups** tab.
1. Click the **Bronze** consumer group you just created.
1. Click **Add Configuration**.
1. In the dialog, complete only the following fields with the following parameters:
    1. Window Size: 30
    1. Window Type: Sliding
    1. Limit: 6
    1. Retry After Jitter Max: 0
1. Click **Save**.

> 

Reviewer: As a tutorial I think this is missing a wrap-up of what was completed and where a user can go from here.

Reviewer: @cloudjumpercat Just remove the conclusion heading and I think it will be fine.

---
title: Create Consumer Groups in Konnect
content_type: tutorial
badge: enterprise
---
With consumer groups, you can define any number of rate limiting tiers and
apply them to subsets of consumers, instead of managing each consumer
individually.

For example, you could define three consumer groups:
* **gold tier** with 1000 requests per minute
* **silver tier** with 10 requests per second
* **bronze tier** with 6 requests per 30 seconds

The `consumer_groups` endpoint works together with the [Rate Limiting Advanced plugin](/hub/kong-inc/rate-limiting-advanced).

Consumers that are not in a consumer group default to the Rate Limiting advanced
plugin’s configuration, so you can define tier groups for some users and
have a default behavior for consumers without groups.

To use consumer groups for rate limiting, you must:
* Create a consumer group and create or add consumers to it
* Configure the Rate Limiting Advanced plugin globally with the `enforce_consumer_groups`
and `consumer_groups` parameters, setting up the list of consumer groups that
the plugin accepts
* Configure a rate limiting policy for each consumer group, overriding the 
plugin's global configuration

> 

Reviewer: If you introduce this like
"This tutorial will walk through creating a bronze tier consumer group...:"
